### PR TITLE
Add support for 'podman top' on FreeBSD

### PR DESCRIPTION
**Pull request by dfr**
This is a simplistic implementation which runs ps(1) on the host and does not try to support 'podman pod top' - I'll tackle that in a separate PS.

```release-note
None
```

### CODE REVIEW

File: libpod/container_top_freebsd.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	"github.com/google/shlex"
+)
+
+var isDescriptor = map[string]bool{}
+
+func init() {
+	allDescriptors, err := util.GetContainerPidInformationDescriptors()
+	if err != nil {
+		// Should never happen
+		return
```
Comment (mheon) [inline libpod/container_top_freebsd.go:25]: Probably at least worth a logged error?

Comment (dfr) [reply libpod/container_top_freebsd.go:25]: Good point

**Review by dfr — COMMENTED** [review]
This is effectively dead code - the implementation changed to call (*Container).jailName. I'll remove it.

**Comment by rhatdan** [comment]
LGTM
@containers/podman-maintainers PTAL

**Comment by Luap99** [comment]
Question, how do plan to document this? https://github.com/containers/podman/blob/main/docs/source/markdown/podman-top.1.md mentions descriptors that are not usable for on bsd.

**Comment by dfr** [comment]
> Question, how do plan to document this? https://github.com/containers/podman/blob/main/docs/source/markdown/podman-top.1.md mentions descriptors that are not usable for on bsd.

Thats a good question - I am hoping to be able to use markdown-preprocess in some way to help with this, perhaps extending @@option and/or @@include to allow platform-specific includes. This could also be used to strip out linux-only flags (e.g. cgroup related) from the FreeBSD manpages.

In the short term, 'podman top --help' does list the set of supported descriptors - I will probably expand that list to include the full list from FreeBSD's ps(1) manpage.

**Comment by rhatdan** [comment]
I think the options/include is a good idea.  Perhaps support a .ARCH flag. For example
```
device-read-bps.md
device-read-bps.md.freebsd
```
On all non freebsd platforms, it reads the default.  On freebsd it reads the specific version.  If file is empty then the option does not get added.

Similarly we could handle this with podman-remote.1.md versus podman.1.md

@edsantiago WDYT?

**Comment by edsantiago** [comment]
My gut reaction is a literal "oof", like a punch in the gut. I see that as introducing high complexity and high risk for very little benefit. Couldn't we just add a human-readable "on FreeBSD, this-and-that" note to the existing documents?

**Comment by dfr** [comment]
> My gut reaction is a literal "oof", like a punch in the gut. I see that as introducing high complexity and high risk for very little benefit. Couldn't we just add a human-readable "on FreeBSD, this-and-that" note to the existing documents?

I certainly don't want to add friction or risk for this. For now, I'm not focusing on the manpages at all while I am working through adding functionality but when I return to it, I'll try to keep things as simple as possible.

**Comment by mheon** [comment]
I'm going to go ahead and merge this, given we're deferring manpage work.

/lgtm

**Pull request merged**